isis/src/mro/apps/ctxcal/main.cpp

```diff
@@ -139,21 +140,30 @@ void IsisMain() {
   //    iof = conversion factor from counts/ms to i/f
   bool convertIOF = ui.GetBoolean("IOF");
   if(convertIOF) {
-    // Get the distance between Mars and the Sun at the given time in
-    // Astronomical Units (AU)
-    QString bspKernel = p.MissionData("base", "/kernels/spk/de???.bsp", true);
-    furnsh_c(bspKernel.toLatin1().data());
-    QString satKernel = p.MissionData("base", "/kernels/spk/mar???.bsp", true);
-    furnsh_c(satKernel.toLatin1().data());
-    QString pckKernel = p.MissionData("base", "/kernels/pck/pck?????.tpc", true);
-    furnsh_c(pckKernel.toLatin1().data());
-    double sunpos[6], lt;
-    spkezr_c("sun", etStart, "iau_mars", "LT+S", "mars", sunpos, &lt);
-    double dist1 = vnorm_c(sunpos);
-    unload_c(bspKernel.toLatin1().data());
-    unload_c(satKernel.toLatin1().data());
-    unload_c(pckKernel.toLatin1().data());
-
+    double dist1 = 1; 
+    try {
+      Camera *cam;
+      cam = icube->camera();
+      cam->setTime(startTime);
+      dist1 = cam->sunToBodyDist();
+    }
+    catch(IException &e) { 
+      // Get the distance between Mars and the Sun at the given time in
+      // Astronomical Units (AU)
+      QString bspKernel = p.MissionData("base", "/kernels/spk/de???.bsp", true);
+      furnsh_c(bspKernel.toLatin1().data());
+      QString satKernel = p.MissionData("base", "/kernels/spk/mar???.bsp", true);
+      furnsh_c(satKernel.toLatin1().data());
+      QString pckKernel = p.MissionData("base", "/kernels/pck/pck?????.tpc", true);
+      furnsh_c(pckKernel.toLatin1().data());
+      double sunpos[6], lt;
+      spkezr_c("sun", etStart, "iau_mars", "LT+S", "mars", sunpos, &lt);
+      dist1 = vnorm_c(sunpos);
+      unload_c(bspKernel.toLatin1().data());
+      unload_c(satKernel.toLatin1().data());
+      unload_c(pckKernel.toLatin1().data());
```

Comment:
It would be good to add a NaifStatus::CheckErrors() call here to check for failures in any of these SPICE calls. See #4038  #4115 for why this matters.

Reply:
Only one NaifStatus check is necessary. Just surround all the spicelib call with one. It will throw on the first one.

Reply:
Per our conversation, this looks good.